src/control.cc

```diff
@@ -37,34 +42,34 @@ bool control_conn_t::process_packet()
     // returns false it has either deleted the connection or marked it for deletion; we
     // shouldn't touch instance members after that point.
 
-    int pktType = rbuf[0];
-    if (pktType == DINIT_CP_QUERYVERSION) {
+    cp_cmd pktType = (cp_cmd)rbuf[0];
+    if (pktType == cp_cmd::QUERYVERSION) {
         // Responds with:
-        // DINIT_RP_CVERSION, (2 byte) minimum compatible version, (2 byte) actual version
-        char replyBuf[] = { DINIT_RP_CPVERSION, 0, 0, 0, 0 };
+        // cp_rply::CVERSION, (2 byte) minimum compatible version, (2 byte) actual version
+        char replyBuf[] = { (cp_rply_t)cp_rply::CPVERSION, 0, 0, 0, 0 };
```

Comment:
Casts should be to the destination type, which would be `char` here, if needed at all.

`(cp_rply_t)cp_rply::CPVERSION` is superfluous - the type of `cp_rply::CPVERSION` is already `cp_reply_t`.

Reply:
I think a cast is actually necessary here, because cp_rply::CPVERSION is of type cp_rply and not cp_rply_t, as it's an enum class.

I've looked it up and narrowing is also applied on static variables in initializer lists. The difference between casting to char or cp_rply_t type in this case would mostly be, that the compiler complains as soon as the value would be out of range for the target type (char, so >127 in this case). So I agree casting to (char) is probably best here.

Reply:
Right, you're correct, I misread it - still, casting to char is probably best, as you say.

Reply:
I've changed all relevant casts to (char).